### ISSUE TITLE
Timeout in StackExchange.Redis version 2 (after migrating from 1.2.6)

### ISSUE THREAD

**Issue by sterchelen** [StackExchange/StackExchange.Redis#1017]
Hi,

Dotnet version : Dotnet core 2.1
Since we've upgraded to the latest version (2.0.519) we are facing timeout :
```
A timeout occured while trying to access a key (Timeout SIsMember CP_EMPTY_IMAGE/3204782785) in Redis - 
Timeout awaiting response (60496ms elapsed, timeout is 60000ms), inst: 0, qs: 13, in: 2596, serverEndpoint: 10.240.0.6:6379, mgr: 10 of 10 available, clientName: bidder-eu-84c84d6449-p7llc, 
IOCP: (Busy=0,Free=1000,Min=4,Max=1000), 
WORKER: (Busy=160,Free=32607,Min=4,Max=32767), v: 2.0.519.65453  StackExchange.Redis.RedisTimeoutException: Timeout awaiting response (60496ms elapsed, timeout is 60000ms), inst: 0, qs: 13, in: 2596, serverEndpoint: 10.240.0.6:6379, mgr: 10 of 10 available, clientName: bidder-eu-84c84d6449-p7llc
IOCP: (Busy=0,Free=1000,Min=4,Max=1000)
WORKER: (Busy=160,Free=32607,Min=4,Max=32767), v: 2.0.519.6545
``` 
Roll backing to the old one (1.2.6) resolve timeout issue. 

Any ideas ?

**Comment by darcon77** [comment]
There seems to be a widespread issues with 2.0. Most likely these timeouts are caused by the fact that CPU utilization using Version 2.* of StackExchange.Redis is 100%.

Any update on this?

**Comment by mgravell** [comment]
The combined improvements in 2.0.588 and 2.0.571 should, we believe, address a wide range of timeout and performance problems. I'm closing all timeout issues as "believed fixed in 2.0.588"; if you still see problems, please report them as new issues so we can start investigation with fresh eyes.
